api/specs/webserver/scripts/openapi_projects_metadata.py

```diff
@@ -0,0 +1,63 @@
+""" Helper script to automatically generate OAS
+
+This OAS are the source of truth
+"""
+
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+# pylint: disable=too-many-arguments
+
+
+from enum import Enum
+from typing import Annotated
+
+from _common import CURRENT_DIR, create_openapi_specs
+from fastapi import Depends, FastAPI, status
+from models_library.api_schemas_webserver.projects_metadata import (
+    ProjectCustomMetadataGet,
+    ProjectCustomMetadataReplace,
+)
+from models_library.generics import Envelope
+from simcore_service_webserver.projects._metadata_handlers import (
+    ProjectCustomMetadataGet,
+    ProjectPathParams,
+)
+
+app = FastAPI(redoc_url=None)
+
+TAGS: list[str | Enum] = ["project"]
+
+
+#
+# API entrypoints
+#
+
+
+@app.get(
+    "/projects/{project_id}/metadata/custom",
+    response_model=Envelope[ProjectCustomMetadataGet],
+    tags=TAGS,
+    operation_id="get_project_custom_metadata",
+    status_code=status.HTTP_200_OK,
+)
+async def get_project_custom_metadata(params: Annotated[ProjectPathParams, Depends()]):
+    ...
+
+
+@app.put(
```

Comment:
Q: In the article you link to the PR, they say that you should have only GET and UPDATE (i.e. patch). Here you actually use replace instead. what is the rationale? That means you are forced to pass the whole data to update the resource. Is this not a waste of network/resources?
I understand you currently have only 1 entry to change so PATCH would be equivalent to PUT, but then why not following the standard?

Reply:
Right. I also have doubts about it. 
In reality, it is a full replacement of the metadata but for some reason everybody is calling it "update" and using PATCH. 

Say  you have a project's subresource ``metadata = {"x": 1, "y":2}``. This is a singleton sub-resource and its lifetime is attached to its parent i.e. the project. Therefore you cannot create/delete metadata but only get or "update" . Nonetheless, the concept of update only allows to fully modified the metadata e.g. I would not be able to modify only "y" entry and keep "x" without **replacing** the entire metadata subresource.

Therefore on the one hand, since you cannot create it, update (i.e. PATCH) operation sounds like the right option. On the other hand, we cannot do partial modifications but only fully replace the metadata and so it would be a "replace" (i.e. PUT ) operation.

On the other hand, if the result of the GET is {"project_id": ... , "metadta": ...} the PUT operation would strictly speaking have to be able to replace "project_id" ... althought that does not make sense. 

Conclusion: I will change it to a PATCH but it should be clear that the metadata entry cannot have partial modifications but is replaced at is.